docs/docs/more-info/architecture.mdx

```diff
@@ -0,0 +1,83 @@
+
+An overview of the architecture is displayed below. Following that we describe how these components all works together in the sections that follow.
+
+<div class="text--center">
+  <img alt="Package architecture" src={useBaseUrl('img/instant-openhie-arch.png')} />
+</div>
+
+## Go executable CLI app
+
+The CLI app will allow the setup and configuration of Instant OpenHIE to be easily managed by the user. It is the only executable that the user would need to download to interact with instant OpenHIE. It is written in Go so that it may compile to a platform-independent executable with 0 dependencies. It will know how to fetch the execution image from docker hub and be able to execute it at the command of the user.
+
+The app will allow the user to execute it as an interactive CLI tool, however, it will also have the ability to launch a Web UI and API server that work together to provide the user with a Web interface for managing and executing Instant OpenHIE. Choices made in the Web UI will be sent to the API server and from there the server will issue commands by running the execution image with a particular set of parameters as shown in the diagram.
+
+The CLI app will also be responsible for ensuring the execution image is executed with the correct configuration to ensure it works cross-platform and with the necessary user setting. For example: it will mount the docker socket file into the container so that docker container are created on the host and it will mount various user config files for Kubernetes, docker hub and/or AWS credentials. This enables the execution image to execute as if it were configured like the users host system.
+
+## The entry point script
+
+The execution image, when executed itself, runs an execution script that is written in Typescript. The purpose of this script is to accept various parameters via the command line, discover the available packages and execute the requested action for a list of the available packages. The list of commandline options that it supports are shown as an example in the architecture diagram. This script is the heart of Instant OpenHIE as it is what actually executes and controls the packages. When the execution image is run this is the script that gets executed to perform all actions.
+
+The execution image bundles all the dependencies that the execution script requires so that when the CLI app downloads the execution image from docker hub it is ready to go. The user will be require to have docker installed to run this image. The bundled dependencies in this image include: the docker and docker-compose clients, kubectl and the cucumber executable for testing purposes.
```

Comment:
```suggestion
The execution image bundles all the dependencies that the execution script requires so that when the CLI app downloads the execution image from docker hub it is ready to go. The user will be required to have docker installed to run this image. The bundled dependencies in this image include: the docker and docker-compose clients, kubectl and the cucumber executable for testing purposes.
```